viewing_party/party.py

```diff
@@ -0,0 +1,135 @@
+    if len(user_data["watched"]) < 1:
+        return 0
+    else:
+        total_ratings = 0
+        for friend_watched in user_data["watched"]:
+            total_ratings += friend_watched["rating"]
+    return total_ratings/len(user_data["watched"])
+
+def get_most_watched_genre(user_data):
+    popular_genre = {}
+    for friend_watched in user_data["watched"]:
+        if friend_watched["genre"] not in popular_genre:
+            popular_genre[friend_watched["genre"]] = 1
+        else:
+            popular_genre[friend_watched["genre"]] += 1
+    # Question for later
+    # max_key = max(popular_genre, key=popular_genre.get)
+    # return max_key
+    max_count = 0
+    pop_genre = None
+    for genre in popular_genre:
+        count = popular_genre[genre]
+        if count > max_count:
+            max_count = count
+            pop_genre = genre
+    return pop_genre
+
+def get_unique_watched(user_data):
+  options_to_watch = []
+  friends_watched = []
+  for friend_watched in user_data["friends"]:
+    for movie_watched in friend_watched["watched"]:
+        if movie_watched not in friends_watched:
+            friends_watched.append(movie_watched)
+
+  for user_watched in user_data["watched"]:
+      if user_watched not in friends_watched:
+        options_to_watch.append(user_watched)
+  return options_to_watch
+
+def get_friends_unique_watched(user_data):
+    options_to_watch = []
+    user_already_watched = []
+    for user_watched in user_data["watched"]:
+        user_already_watched.append(user_watched)
+    print(user_already_watched)
+    
+    for friend_watched in user_data["friends"]:
+        for movie_watched in friend_watched["watched"]:
+            if movie_watched not in user_already_watched and \
+                movie_watched not in options_to_watch:
+
+                options_to_watch.append(movie_watched)
+    return options_to_watch
+
+def get_available_recs(user_data):
+  friend_recs = get_friends_unique_watched(user_data)
+  recommended = []
+
+  for movie in friend_recs:
+    if movie["host"] in user_data["subscriptions"]:
+      recommended.append(movie)
+      
+  return recommended
+
+def get_new_rec_by_genre(user_data):
+  user_pop_genre = ""
+  popular_genre = {}
+  recommended = []
+
+  for user_watched in user_data["watched"]:
+    if user_watched["genre"] not in popular_genre:
+      popular_genre[user_watched["genre"]] = 1
+    else:
+      popular_genre[user_watched["genre"]] += 1
+
+  max_count = 0
+  pop_genre = None
+  for genre in popular_genre:
+    count = popular_genre[genre]
+    if count > max_count:
+      max_count = count
+      pop_genre = genre
+
+  user_pop_genre = pop_genre
+
+  for friend_watched in user_data["friends"]:
+    for friends_watched_movie in friend_watched["watched"]:
+      if friends_watched_movie not in user_data["watched"] and \
+          friends_watched_movie["genre"] == user_pop_genre:
+        recommended.append(friends_watched_movie)
+  return recommended
```

Comment:
even though this works well, how could you have used `get_friends_unique_watched(user_data)` and/or `get_most_watched_genre(user_data)` to help you out?